client/src/components/Sub_RelatedItems/OutfitContainer.jsx

```diff
@@ -1,54 +1,108 @@
-import React, { useState, useEffect } from 'react';
+import React, { useState, useEffect, useRef } from 'react';
+import { BiChevronLeftCircle, BiChevronRightCircle } from 'react-icons/bi';
+import { flushSync } from 'react-dom';
 import SingleProd from './SingleProd.jsx';
 
 export default function OutfitContainer({
   parentProduct, setParentProductID,
   parentProductStyle, currentParentProductStyle,
   parentReviewMetadata, setAsNewOverview, setAllowCardClick,
 }) {
+  // =================== STATES ===================
+  const [myOutfits, setMyOutfits] = useState([]);
+  const [checkStyles, setCheckStyles] = useState({});
+  const [index, setIndex] = useState(0);
+  const cardsRef = useRef(null);
 
-  // const { id } = thisProduct;
+  // =================== EFFECTS ===================
+  useEffect(() => {
+    if (localStorage.length) {
+      const storedCheckStyles = JSON.parse(localStorage.getItem('inUse'));
 
-  // const [myOutfits, setMyOutfits] = useState(new Set());
+      const storedKeys = Object.keys(localStorage);
+      const checkStyleIdx = storedKeys.indexOf('inUse');
 
-  const [myOutfits, setMyOutfits] = useState([]);
-  const [checkStyles, setCheckStyles] = useState({});
+      console.log('\nstoredCheckStyles: ', storedCheckStyles);
+      console.log('\nstoredKeys: ', storedKeys, '\ncheckStyleIdx: ', checkStyleIdx);
 
-  // const [outfitsUpdated, setOutfitsUpdated] = useState(false);
+      storedKeys.splice(checkStyleIdx, 1);
 
-  useEffect(() => {console.log(checkStyles)}, [myOutfits]);
+      const storedOutfits = storedKeys.map((key) => (
+        JSON.parse(localStorage.getItem(key))));
+
+      setMyOutfits(storedOutfits);
+      setCheckStyles(storedCheckStyles);
+    }
+  }, [localStorage]);
+
+  // =================== HELPERS ===================
+  const handleRightClick = () => {
+    flushSync(() => {
+      if (index < myOutfits.length - 1) {
+        setIndex(index + 1);
+      } else {
+        setIndex(0);
+      }
+    });
+    cardsRef.current.scrollIntoView({
+      behavior: 'smooth',
+      block: 'nearest',
+      inline: 'center',
+    });
+  };
+
```

Comment:
wow i didn't realized that useRef can be used in this way!

Reply:
need to look into flushSync. this looks really cool